### PR TITLE
add emoji-picker v1.0 w/ git auto-update

### LINKED ISSUE #5570: [Request] Add emoji-picker
**Library name:**  emoji-picker
**Git repository url:** https://github.com/one-signal/emoji-picker
**npm package url(optional):** 
**Official homepage:** http://one-signal.github.io/emoji-picker/
**Wanna say something? Leave message here:** Add a slick emoji selector to input fields and textareas on your website

=====================
Notes from cdnjs maintainer:
You are welcome to add a library via sending pull request,
it'll be faster then just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by maruilian11**
**git repo url:** https://github.com/one-signal/emoji-picker
**Watch 6
  Star 136
 Fork 9**
@LeaYeh  could u help me check this PR for #5570 ? thank you!
btw, i only add three files under `img` because only those files are used in codes.
- [x] Not found on cdnjs repo
- [x] No already exist issue and PR
- [x] Notable popularity
- [x] Project has public repository on famous online hosting platform (or been hosted on npm) 
- [x] Has valid tags for each versions (for git auto-update)
- [x] Auto-update setup
- [x] Auto-update target/source is valid.
- [x] Auto-update filemap is correct

### CODE REVIEW

**Comment by LeaYeh** [comment]
i'm not very sure that whether need to add `/css` and `/js` into `autoupdate.files` field
@cdnjs/intern

**Comment by maruilian11** [comment]
@Amomo @Piicksarn we need your advice.

**Comment by Amomo** [comment]
No, `/css` and `/js` folder is for its web site pages.

BTW, as its [Installation & Usage](https://github.com/one-signal/emoji-picker#installation--usage) said, we also need to add ...
- lib/css/nanoscroller.css
- lib/js/nanoscroller.min.js
- lib/js/tether.min.js
- lib/js/config.js
- lib/js/util.js
- lib/js/jquery.emojiarea.js

**Comment by maruilian11** [comment]
@Amomo thanks for your recommend.

**Comment by maruilian11** [comment]
@Amomo does your "No" means not add `css/emoji*.+(css|map)` and `js/emoji-picker*.+(js|map)`?

**Comment by LeaYeh** [comment]
@Amomo how to know `/css` and `/js` folder is for its web site pages?

**Comment by maruilian11** [comment]
@Amomo please take a look~ thank you~~

**Comment by Amomo** [comment]
@LeaYeh @maruilian11 You can see source code of their website~
They use files in /css and /js.
You can also see their [installation & usage](https://github.com/one-signal/emoji-picker#installation--usage), we should host files under `lib/`.

**Comment by maruilian11** [comment]
@LeaYeh could you please check this PR again? thank you

**Comment by LeaYeh** [comment]
i think this pr is ok.
@PeterDaveHello

**Comment by PeterDaveHello** [comment]
Don't you need `emoji_spritesheet_*.png` under img folder?

**Comment by maruilian11** [comment]
@PeterDaveHello but i found they didn't be used in code.

**Comment by PeterDaveHello** [comment]
Seems it's belongs to https://github.com/diy/jquery-emojiarea, which doesn't have a version tag in its repo, and according to emoji-picker's readme, we may need to use them together, so I think we should add them both in this PR before jquery-emojiarea has a valid version and been added on cdnjs, maybe you can create an issue ticket for that.

**Comment by LeaYeh** [comment]
do we also need to add [nanoScrollerJs](https://github.com/one-signal/emoji-picker#credits) like jquery-emojiarea in this pr?

**Comment by PeterDaveHello** [comment]
Do it this time.

**Comment by maruilian11** [comment]
@LeaYeh i have add those `.png`. could you help me check it again? thank you.

**Comment by LeaYeh** [comment]
i think it is ok, you just need to rebase it

**Comment by maruilian11** [comment]
@Amomo could you help me to check this PR? thank you!

**Comment by Amomo** [comment]
@maruilian11 it looks great, but it would be nicer if you tag the author in your commit message.
Could you give it a modification?

**Comment by maruilian11** [comment]
@Amomo i have taged the author in commit message, could you check the PR again? thanks~

**Comment by Amomo** [comment]
@PeterDaveHello I think this PR is fine, could you take a look? thank you! :-D

**Comment by PeterDaveHello** [comment]
Did you guys read its readme?
https://github.com/one-signal/emoji-picker#installation--usage

We didn't host individual jquery.emojiarea, so why not add it in?

**Comment by maruilian11** [comment]
@PeterDaveHello you mean send a new PR to add  jquery.emojiarea? or add it in this lib?

**Comment by maruilian11** [comment]
i see.

**Comment by maruilian11** [comment]
@PeterDaveHello i have add `jquery.emojiarea.js`, please take a look.

**Comment by PeterDaveHello** [comment]
@maruilian11 So what about the others? Do you not need to confirm?

**Comment by maruilian11** [comment]
@Amomo i have added `js/util.js`, could you help me check again? thank you!

**Comment by Amomo** [comment]
Don't we need `"js/tether.min.js"` and `"js/config.js"`?

**Comment by maruilian11** [comment]
@PeterDaveHello i thought `tether.min.js` and `nanoscroller.min.js` are belong to other lib, could i just add `config.js`, `util.js`, `jquery.emojiarea.js` and `emoji-picker.js` under `lib/js`
https://github.com/one-signal/emoji-picker#installation--usage

**Comment by PeterDaveHello** [comment]
Yes, you should make sure that two libs been hosted on cdnjs and have corresponding version to the lib's dependency.

**Comment by maruilian11** [comment]
@PeterDaveHello cdnjs do host those two libs, but  emoji-picker doesn't list them in `dependency`. so do i just send a PR to add them to dependency of emoji-picker?

**Comment by PeterDaveHello** [comment]
@maruilian11 why don't you take a look what's the version in the  repo? At least we know what's the version been referenced.

**Comment by maruilian11** [comment]
i'm can't tell `tether.min.js`'s version by minified file, do i need to ask the author?

**Comment by PeterDaveHello** [comment]
@maruilian11 if it's officially minified, you can compare its md5sum with cdnjs hosted versions

**Comment by maruilian11** [comment]
@PeterDaveHello i have confirmed. 
> Yes, you should make sure that two libs been hosted on cdnjs and have corresponding version to the lib's dependency.
@Amomo could you help me check this PR again? thank you!

**Comment by maruilian11** [comment]
ping @Amomo

**Comment by Amomo** [comment]
@maruilian11 sorry for late. > <
Could you please rebase this PR on latest master branch again?
Thank you very much!

**Comment by maruilian11** [comment]
@Amomo it's ok~ i have rebased this branch upon our latest master.

**Comment by Amomo** [comment]
@maruilian11 sorry, could you do it again? thanks...😓

**Comment by maruilian11** [comment]
@Amomo it's ok~

**Comment by PeterDaveHello** [comment]
Rebase and what? @Amomo I didn't see your review result.

**Comment by maruilian11** [comment]
ping @Amomo

**Comment by maruilian11** [comment]
@PeterDaveHello could you give me a review result? then i will rebase the pr upon our latest master.

File: ajax/libs/emoji-picker/package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    "jquery-plugin"
+  ],
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/one-signal/emoji-picker.git",
+    "basePath": "lib",
+    "files": [
+      "css/emoji*.+(css|map)",
+      "img/IconsetSmiles*.png",
+      "img/blank*.gif",
```
Comment (PeterDaveHello) [inline ajax/libs/emoji-picker/package.json]: Use `"img/*.+(gif|png)"` here.

File: ajax/libs/emoji-picker/package.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      "js/emoji-picker*.+(js|map)",
+      "js/jquery.emojiarea*.js",
+      "js/util*.js",
+      "js/config*.js"
+    ]
+  },
+  "requiredFiles": [
```
Comment (PeterDaveHello) [inline ajax/libs/emoji-picker/package.json]: Drop `"requiredFiles"` as we talked recently.

File: ajax/libs/emoji-picker/package.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  "author": {
+    "name": "Jason Pang @ OneSignal",
+    "url": "https://onesignal.com"
+  },
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/one-signal/emoji-picker"
+  },
+  "license": "MIT",
+  "homepage": "https://github.com/one-signal/emoji-picker",
```
Comment (PeterDaveHello) [inline ajax/libs/emoji-picker/package.json]: Should be `https://one-signal.github.io/emoji-picker/`

**Comment by PeterDaveHello** [comment]
Don't forget to rebase the branch for this PR upon our latest master branch, thanks.

**Comment by maruilian11** [comment]
@PeterDaveHello update with map files.

File: ajax/libs/emoji-picker/package.json [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    "icons",
+    "jquery-plugin"
+  ],
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/one-signal/emoji-picker.git",
+    "basePath": "lib",
+    "files": [
+      "img/*.(png|gif)",
+      "+(css|js)/+(emoji*|jquery.emojiarea*|util*|config*).+(css|js|map)"
```
Comment (PeterDaveHello) [inline ajax/libs/emoji-picker/package.json]:
```diff
- "+(css|js)/+(emoji*|jquery.emojiarea*|util*|config*).+(css|js|map)"
+ "+(css|js)/+(emoji|jquery.emojiarea|util|config)*.+(css|js|map)"
```

**Comment by maruilian11** [comment]
@PeterDaveHello updated.

**Comment by PeterDaveHello** [comment]
There is only 1 git tag - `1.0` but you are using `1.0.0`, that's not good.

File: ajax/libs/emoji-picker/package.json [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+    "emoticons",
+    "icons",
+    "jquery-plugin"
+  ],
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/one-signal/emoji-picker.git",
+    "basePath": "lib",
+    "files": [
+      "img/*.(png|gif)",
```
Comment (PeterDaveHello) [inline ajax/libs/emoji-picker/package.json]: `"img/*.+(png|gif)",`

**Comment by maruilian11** [comment]
@PeterDaveHello  updated.

**Comment by PeterDaveHello** [comment]
You are still using 1.0.0 as its version, is there a problem? I thought it's very simple ...

**Comment by maruilian11** [comment]
@PeterDaveHello updated.

**Pull request merged**